File: hash/programmers_42578/solution.py
def solution(clothes):
    hash_table = {}
    answer = 1
    
    for name, category in clothes:
        if category in hash_table:
            hash_table[category] += 1
        else:
            hash_table[category] = 1
    
    for key in hash_table:
        answer *= (hash_table[key] + 1)
    
    return answer - 1

File: hash/programmers_42578/solution.js
function solution(clothes) {
    const hashTable = {}
    let answer = 1;
    
    clothes.forEach(cloth => {
        const [name, category] = cloth;
        
        if(hashTable.hasOwnProperty(category)){
            hashTable[category] ++;
        } else {
            hashTable[category] = 1;
        }
    })
    
    for(const key in hashTable) {
        answer *= (hashTable[key] + 1);
    }
    
    return answer - 1;
}